modules/compatibility/src/test/java/org/apache/ignite/compatibility/sql/SqlQueryRegressionsTest.java

```diff
@@ -0,0 +1,359 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compatibility.sql;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.compatibility.sql.model.City;
+import org.apache.ignite.compatibility.sql.model.Company;
+import org.apache.ignite.compatibility.sql.model.Country;
+import org.apache.ignite.compatibility.sql.model.Department;
+import org.apache.ignite.compatibility.sql.model.ModelFactory;
+import org.apache.ignite.compatibility.sql.model.Person;
+import org.apache.ignite.compatibility.sql.randomsql.RandomQuerySupplier;
+import org.apache.ignite.compatibility.sql.randomsql.Schema;
+import org.apache.ignite.compatibility.sql.randomsql.Table;
+import org.apache.ignite.compatibility.sql.runner.PredefinedQueriesSupplier;
+import org.apache.ignite.compatibility.sql.runner.QueryDuelBenchmark;
+import org.apache.ignite.compatibility.sql.runner.QueryDuelResult;
+import org.apache.ignite.compatibility.testframework.junits.Dependency;
+import org.apache.ignite.compatibility.testframework.junits.IgniteCompatibilityAbstractTest;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.ClientConnectorConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgnitionEx;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.G;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.lang.IgniteInClosure;
+import org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi;
+import org.apache.ignite.spi.discovery.tcp.ipfinder.TcpDiscoveryIpFinder;
+import org.apache.ignite.spi.discovery.tcp.ipfinder.vm.TcpDiscoveryVmIpFinder;
+import org.apache.ignite.testframework.junits.multijvm.IgniteProcessProxy;
+import org.junit.Test;
+
+/**
+ * Test for SQL queries regressions detection.
+ * It happens in the next way:
+ * 
+ * 1. Test starts two different Ignite versions: current version and the old one.
```

Comment:
"the old one" means some specific old version. Seems, you mean "one of previous versions".
So, how old version will be chosen? 
Would you please give few words on this.

Reply:
It should be passed as System property. See https://github.com/gridgain/gridgain/pull/1296/files#diff-5d8bc8f077dd7b97c909e7fef5b24756R238

Reply:
Would you please describe here how to run the test? or refer to sys properties? as it is still unclear how  "old version" is evaluated.